### DISCUSSION TITLE
Cheat: allow all track pieces

### DISCUSSION THREAD

**Discussion by jensj12** [OpenRCT2/OpenRCT2#15711] [Feature requests]
There are several tricks that involve using track pieces from different rides types on the same ride (the teleporter rides that use golf holes for example). Using the 'allow arbitrary ride type changes' cheat is required for this, but swapping between rides is tedious to me (especially if you don't know the order of the rides or the numbers associated with them). That's why I'd like to suggest a new cheat:

The cheat removes all ride-spicific limitations on track pieces and allows any track piece (golf holes, vertical track, sharp curves, flat to steep with long base) to be used on any ride.

While it may look easy to implement, the ride construction window isn't ready for this as it doesn't allow all curve types to be shown at the same time and only allows one type of flat to steep pieces.

**Comment by Gymnasiast** [comment]
Well, I'm glad you included that last paragraph, as I would certainly have said that otherwise. In addition, we cannot just use the dropdown either: it would take too much screen space and dropdowns can only handle 64 items anyway.

If anyone has a good idea how to handle this, please comment. Or even better: if anyone can implement it, please open a PR.

**Comment by rwjuk** [comment]
Just throwing ideas around here, I'm not sure whether it would work, or even if it's a *good* idea - how about a dropdown at the bottom of the ride construction window allowing ride type changes? This way you wouldn't need to load all pieces at once, but users can make the change without having to go the long way around.

**Comment by IntelOrca** [comment]
@rwjuk This is exactly what I want for changing the track type mid coaster. Although that would be when you could have multiple track types for a single ride.

**Comment by jensj12** [comment]
If that works, why isn't that used for the current UI to switch ride types? Opening an extra window isn't the hard part, it's the spinner that makes it tedious. Opening a new window with a list of ride types would work as well.

When building vertical track, all the slope options are shifted a spot to the left or to the right. So idea: add 'spinner' buttons (arrow left/right) on both sides of the window to do that manually, both for the slope and for the curve. Or automatically shift it so that the currently selected piece is in the center.

Edit: just looked through the code and let is_track_enabled always return true. Now I see how some unusual track pieces are just hacked in. Both the curve/slope selection and the special track dropdown need a lot of work to get it working properly.
